application/features/bootstrap/MfaUnitTestsContext.php

```diff
@@ -49,19 +57,84 @@ protected function createNewUserInDatabase($username)
         return $user;
     }
 
+    protected function createMfa($type, $verified=1, $user=null)
+    {
+        if ($user ===null) {
+            $user = $this->tempUser;
+        }
+        $mfa = new Mfa();
+        $mfa->user_id = $user->id;
+        $mfa->type = $type;
+        $mfa->verified = $verified;
+
+        Assert::true($mfa->save(), "Could not create new mfa.");
+        $user->refresh();
+        $this->mfaId = $mfa['id'];
+        $this->mfa = Mfa::findOne(['id' => $mfa['id']]);
+    }
+
     /**
      * @Given I have a user with a backup codes mfa option
      */
     public function iHaveAUserWithABackupCodesMfaOption()
     {
-        $user = $this->createNewUserInDatabase('mfa_tester');
-        $mfaCreateResult = Mfa::create($user->id, Mfa::TYPE_BACKUPCODE);
-
-        $this->mfaId = $mfaCreateResult['id'];
-        Assert::notEmpty($this->mfaId);
+        $this->tempUser = $this->createNewUserInDatabase('mfa_tester');
+        $this->createMfa(Mfa::TYPE_BACKUPCODE);
         MfaBackupcode::deleteAll();
     }
 
+    /**
+     * @Given I have a user with an unverified totp mfa option
+     */
+    public function iHaveAUserWithAnUnverifiedTotpMfaOption()
+    {
+        Mfa::deleteAll([]);
```

Comment:
In case you find this preferable, when using `deleteAll()` to delete all of those records (no conditions), you don't need to pass it any parameters at all.

Reply:
Done.